src/main/scala/latis/util/hapi/parameter.scala

```diff
@@ -43,43 +43,38 @@ final case class ArrayParameter(
   bin: NonEmptyList[Bin]
 ) extends Parameter
 
-object Parameter {
-
-  implicit val decodeParameter: Decoder[Parameter] = new Decoder[Parameter] {
-    final def apply(c: HCursor): Decoder.Result[Parameter] = for {
-      name   <- c.get[String]("name")
-      tyName <- c.get[String]("type")
-      units  <- c.get[Option[String]]("units")
-      // Only time and string parameters have length.
-      length <- if (tyName == "string" || tyName == "isotime") {
-        c.get[Int]("length").map(Option(_))
-      } else Right(None)
-      fill   <- c.get[Option[String]]("fill")
-      size   <- c.get[Option[NonEmptyList[Int]]]("size").flatMap {
-        case Some(s) => Right(Option(s))
+given decodeParameter: Decoder[Parameter] = new Decoder[Parameter] {
```

Comment:
I'm unsure about making this a top level definition.

One downside of doing so is that you need to explicitly import the given in order to use it:

```scala
import latis.util.hapi.Parameter

// Won't work
implicitly[io.circe.Decoder[Parameter]]

import latis.util.hapi.*

// Also won't work
implicitly[io.circe.Decoder[Parameter]]

import latis.util.hapi.given

// Works, but will also import every other given at the latis.util.hapi level
implicitly[io.circe.Decoder[Parameter]]
```

When wrapped in a `Parameter` object, it works without additional imports:

```scala
import latis.util.hapi.Parameter

// Works
implicitly[io.circe.Decoder[Parameter]]
```

I'm thinking about this from the perspective of using this as a library for working with HAPI responses. We didn't have an issue here because everything that uses it is already in `latis.util.hapi`. But if we made the same change for the `Info` decoder, then we'd need to change imports.